extensions/amp-ad-network-adsense-impl/0.1/amp-ad-network-adsense-impl.js

```diff
@@ -31,20 +31,22 @@ import {
   getCorrelator,
 } from '../../../ads/google/a4a/utils';
 import {getLifecycleReporter} from '../../../ads/google/a4a/performance';
-import {documentStateFor} from '../../../src/document-state';
 import {getMode} from '../../../src/mode';
 import {stringHash32} from '../../../src/crypto';
 import {domFingerprintPlain} from '../../../src/utils/dom-fingerprint';
+import {viewerForDoc} from '../../../src/viewer';
 
 /** @const {string} */
 const ADSENSE_BASE_URL = 'https://googleads.g.doubleclick.net/pagead/ads';
 
-/** @const {!Object<string, string>} */
+/**
+ * See `VisibilityState` enum.
+ * @const {!Object<string, string>}
+ */
 const visibilityStateCodes = {
```

Comment:
That was my question as well :) `preview` is not a valid value that I know of, so not sure where it comes from. Please let me know and send me a source with the info if that was correct in your opinion.